### ISSUE TITLE
LibreSSL exposes misuse of $ENV 

### ISSUE THREAD

**Issue by ppunosevac** [OpenVPN/easy-rsa#76]
I am a heavy easy-rsa user for my OpenVPN gateways but also for my LDAP servers. I also happen to use OpenBSD as my OS of choice.

On Sunday I spent 30 minutes playing with easy-rsa which is shipped broken on OpenBSD 5.8 until I realized what was going on. Our developer sthen has already reverted easy-rsa to OpenSSL run dependency per comment

switch easy-rsa to using openssl to unbreak; libressl doesn't allow
$ENV:: in config files and easy-arrrrsa uses this heavily.

This is the discussion that followed on ports @openbsd. 

http://marc.info/?l=openbsd-ports&m=144578740817527&w=2

To summarize discussion in one sentence I was wondering if you guys can think of the better way  of communicating to libraries instead of using environment variables?

**Comment by ecrist** [comment]
I read through the entire thread and it would be nice for us to support both LibreSSL and PolarSSL in the future.  This is likely a 3.2 fix.

**Comment by mvcube** [comment]
Here is a simple fix to make it work on macOS High Sierra. It consists of a a new template config file libressl.cfg.template and an updated whichopensslcnf script. The idea is to embed the necessary environment variables directly into the config file which is genererated form the template.

[easy-rsa-libressl.zip](https://github.com/OpenVPN/easy-rsa/files/1598032/easy-rsa-libressl.zip)

**Comment by mvcube** [comment]
I don't have write access to the repository so I ask one of the contributors to do this for me.

**Comment by ecrist** [comment]
Should be resolved in 93b0f2e74bb600c7f05c20eaa97f7b12fb7ba670
